build/stf-run-ci/tasks/main.yml

```
@@ -5,29 +5,68 @@
     branch: "{{ working_branch | default('master') }}"
     namespace: "{{ working_namespace | default('service-telemetry') }}"
 
-- name: Setup supporting repositories
-  include_tasks: clone_repos.yml
-
-- name: Setup supporting Operator subscriptions
-  include_tasks: setup_base.yml
-
-- name: Create builds and artifacts
-  include_tasks: create_builds.yml
-  loop:
-    - { name: service-telemetry-operator, dockerfile_path: build/Dockerfile, image_reference_name: sto_image_path, working_build_dir: ../ }
-    - { name: smart-gateway-operator, dockerfile_path: build/Dockerfile, image_reference_name: sgo_image_path, working_build_dir: ./working/smart-gateway-operator }
-    - { name: smart-gateway, dockerfile_path: Dockerfile, image_reference_name: sg_image_path, working_build_dir: ./working/smart-gateway }
-  loop_control:
-    loop_var: artifact
-
-- name: Deploy STF using local artifacts
-  include_tasks: deploy_stf.yml
-
-- name: Validate system is operational
-  shell: ./validate_deployment.sh
-  args:
-    executable: /bin/bash
-  register: validate_deployment
-
-- debug:
-    var: validate_deployment.stdout_lines
+- name: Set default image paths when skipping local builds
```

AFAICT these are only used in the local builds, not at all when skipping them. Should they be moved into the block below?

```suggestion
- name: Set default image paths for local builds
```

---

These are used when skipping local builds, but when the builds have previously been done. The wording I agree is a bit awkward here.

The point of this is that you've done a local build, you've tried a deployment but the Ansible has failed for some reason, but the builds themselves worked fine. This allows you to skip waiting a few minutes for the builds to re-run when the code never changed for the builds.

Primarily this functionality is for the CI developer who wants to make changes in the workflow, test everything, but doesn't want to wait several minutes for local build images to re-build.